auth/auth-router.js

```diff
@@ -1,11 +1,52 @@
+const bcrypt = require('bcryptjs');
+const express = require('express');
+const jwt = require('jsonwebtoken');
+const Users = require('../users/users-model');
+// const userValid = require('../users/user-validation');
+const secrets = require('../config/secrets');
+
 const router = require('express').Router();
 
-router.post('/register', (req, res) => {
-  // implement registration
+
+router.post('/register', async (req, res, next) => {
+  try {
+    const registered = await Users.add(req.body)
+
+    res.status(201).json(registered)
+  } catch (err) {
+    next(err)
+  }
 });
 
-router.post('/login', (req, res) => {
-  // implement login
+
+router.post('/login', async (req, res, next) => {
+  try {
+    const { username, password } = req.body
+    const user = await Users.findByUsername({ username }).first()
+    const passwordValid = await bcrypt.compare(password, user.password)
+
+    if (user && passwordValid) {
```

Comment:
We didn't really do a good job walking through this section to mitigate user errors when logging in.  Your server will crash if it is sent `undefined` variables.  Basically, if username or password is not filled out, you will run into errors validating the password.  `user` will also be undefined and it will cause your server to crash.  You will need If/else statements to account for user errors.